### ISSUE TITLE
'Convert typeof to nameof' has bug with nint and nuint

### ISSUE THREAD

**Issue by jnm2** [dotnet/roslyn#47128]
**Version Used**: 16.8-p2

```cs
using System;

class C
{
    void M()
    {
        // 💡 Convert 'typeof' to 'nameof'
        Console.WriteLine([||]typeof(nint).Name); // IntPtr
    }
}
```

Expected result:

```cs
        Console.WriteLine(nameof(IntPtr)); // IntPtr
```

Actual result:

```cs
        Console.WriteLine(nameof(nint)); // nint
```

Also, I expected a compiler error for `nameof(nint)` like there is for `nameof(int)`.

**Comment by CyrusNajmabadi** [comment]
Tagging as both compiler and IDE.  IDE to def fix the 'convert' feature, compiler to investigate what should `nameof(nint)` do.  Note: i agree with @jnm2 that it shoudl error.  Note that `nameof(dynamic)` does fail, which is the same 'semantically contextual identifier' case that i think `nint` falls into.

**Comment by Youssef1313** [comment]
@CyrusNajmabadi @jnm2 
Probably by design?


#38821 mentions nameof(nint) as valid based on my understanding.

@jcouv Can you confirm if nint is intentionally allowed in nameof?

**Comment by jcouv** [comment]
`nameof(int)` is disallowed, so I would expect the same for `nameof(nint)` when it binds to the native integer type (as opposed to a local variable named "nint").
